src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.tsx

```diff
@@ -14,45 +14,122 @@
  * limitations under the License.
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
 import ProteinDomainImage from 'src/content/app/entity-viewer/gene-view/components/protein-domain-image/ProteinDomainImage';
 import ProteinImage from 'src/content/app/entity-viewer/gene-view/components/protein-image/ProteinImage';
 import ProteinFeaturesCount from 'src/content/app/entity-viewer/gene-view/components/protein-features-count/ProteinFeaturesCount';
+import ExternalReference from 'src/shared/components/external-reference/ExternalReference';
+
+import {
+  ExternalSource,
+  externalSourceLinks
+} from 'src/content/app/entity-viewer/shared/helpers/entity-helpers';
+import { fetchTranscript } from 'src/content/app/entity-viewer/shared/rest/rest-data-fetchers/transcriptData';
+import {
+  fetchProteinSummary,
+  ProteinSummary
+} from 'src/content/app/entity-viewer/shared/rest/rest-data-fetchers/proteinData';
 
 import { Transcript } from 'src/content/app/entity-viewer/types/transcript';
 
 import styles from './ProteinsListItemInfo.scss';
 
 type Props = {
-  transcript: Transcript;
+  transcriptId: string;
   trackLength: number;
 };
 
 const ProteinsListItemInfo = (props: Props) => {
-  const { transcript, trackLength } = props;
+  const { transcriptId, trackLength } = props;
+  const [transcript, setTranscript] = useState<Transcript | null>(null);
+  const [proteinSummary, setProteinSummary] = useState<ProteinSummary | null>(
+    null
+  );
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    fetchTranscript(props.transcriptId, abortController.signal).then(
+      (result) => {
+        if (result) {
+          setTranscript(result);
+        }
+      }
+    );
+
+    fetchProteinSummary(props.transcriptId).then((result) => {
+      if (result) {
+        setProteinSummary(result);
+      }
+    });
```

Comment:
I don't know how temporary this data fetching is, but from what I can see:

1) Abort controller is not even used to prevent updates of an unmounted component
2) In fact, I am not even sure there is particular value in aborting the request itself. What want to avoid is updating the component if the promise happens to be resolved after the component gets unmounted. This can be solved with a simple flag:

```
useEffect(() => {
  const canUpdate = true;

  Promise.all([
    fetchTranscript(props.transcriptId),
    fetchProteinSummary(props.transcriptId)
  ]).then(([transcript, proteinSummary]) => {
    if (!canUpdate) {
      return;
    }
    transcript && setTranscript(transcript);
    proteinSummary && setProteinSummary(proteinSummary)
  });

  return () => canUpdate = false;
};
```

Reply:
I used the abort controller to cleanup the component on unmount. Also, the fetches are queued with `Promise.all`.